### PR TITLE
TASK-2894, TASK- 2895, TASK-2896, TASK-2897/integrate cct service with letter service and display pdf

### PR DESCRIPTION
**Pull request by carolinegeorge-gov**
### Description

* Updated `pdf` `factory` definition  in `db`
* Added updated `pdf` and `letter` mocks to be linked to one another by `referencedId`
* Integrated `cct-service` `getPdf` calls to downloads.
* Moved the existing pdf stream and generation from the `downloads` page to the `cct-service-api` mocks
* Updated  letters `downloads` page to parse returned pdf so it can be viewed and downloaded

### Related Azure Boards Work Items

[AB#2894](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2894), [AB#2895](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2895), [AB#2896](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2896), [AB#2897](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2897)
### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions

1. Starting at Home, go to Letter
2. Click any letter from the list
3. VIew PDF with Download option on the side

### Additional Notes

* @sebastien-comeau @gregory-j-baker Currently running into an issue that on the second time I click a letter link (doesn't matter which way Click letter 1 then letter 2, click letter 1 twice, etc), it just lags. If I click again, it works fine. If I click a third or any subsequent time after, it works fine. Not sure if its some latency or reading issue but it doesn't block functionality in that it refreshes.

### CODE REVIEW

File: frontend/app/mocks/db.ts
Comment (sebastien-comeau) [inline frontend/app/mocks/db.ts:1]: The letters and assoticated pdfs seeding can be done in a loop.

```ts
const numberOfLetters = faker.datatype.number({ min: 10, max: 20 }); // Adjust min and max as needed
for (let i = 0; i < numberOfLetters; i++) {
  // seed data here
}
```

File: frontend/app/services/cct-service.server.ts [hunk 1]
```diff
@@ -4,34 +4,25 @@ import { z } from 'zod';
...
  */
-export const getPdf = moize(createPdfService, { onCacheAdd: () => log.info('Creating new PDF service') });
+export const getCCTService = moize(createCCTService, { onCacheAdd: () => log.info('Creating new CCT service') });
 
-function createPdfService() {
+function createCCTService() {
   const { CCT_API_BASE_URI } = getEnv();
 
-  async function getPdf(referenceId: string) {
+  async function getPdf(referenceId: string | readonly string[]) {
```
Comment (sebastien-comeau) [inline frontend/app/services/cct-service.server.ts:20]:
```suggestion
  async function getPdf(referenceId: string) {
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/mocks/db.ts
Comment (carolinegeorge-gov) [reply frontend/app/mocks/db.ts:1]: Good idea. Fixed: 5d31d79b871e1f72cefe2a1c8def40b33821fb66.

File: frontend/app/services/cct-service.server.ts (hunk 1 above)
Comment (carolinegeorge-gov) [reply frontend/app/services/cct-service.server.ts]: Fixed: 5d31d79b871e1f72cefe2a1c8def40b33821fb66.

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**